lib/resources/aws/aws_ec2_instances.rb

```diff
@@ -0,0 +1,72 @@
+# author: Christoph Hartmann
+class AwsEc2Instances < Inspec.resource(1)
+  name 'aws_ec2_instances'
+  desc 'Verifies settings for an EC2 instance'
+  supports platform: 'aws'
+  example "
+    describe_ec2_instances.where { public_ip_address !=nil } do
+      its('instance_id') { should eq [] }
+    end
+  "
+
+  # TODO: rewrite to avoid direct injection, using AwsPluralResourceMixin
```

Comment:
Have a look at more recent plural resources - aws_security_groups, for example.  They use the AwsPluralResource mixin to handle several things, including resource param validation and testing backend setup.

Reply:
Will do...